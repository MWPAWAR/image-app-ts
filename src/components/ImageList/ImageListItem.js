import { useDispatch } from 'react-redux';
import { getMbsTextByBytes } from '../../utils/get-mbs-by-bytes';
import imageListItemStyles from './ImageListItem.module.css';
import { setSelectedImg } from '../../store/imageUiState/imageUiState';

const ImageListItem = ({ image }) => {
  const dispatch = useDispatch();
  const handleImageClick = () => dispatch(setSelectedImg(image.id));

  return (
    <div onClick={handleImageClick} className={imageListItemStyles.imageListItem}>
      <img
        tabIndex={0}
        src={image.url}
        className={imageListItemStyles.imageListItemImg}
        alt={image.filename}
      />
      <div className={imageListItemStyles.imageListItemName}>{image.filename}</div>
      <div className={imageListItemStyles.imageListItemSize}>{getMbsTextByBytes(image.sizeInBytes)}</div>
    </div>
  );
};

export default ImageListItem;
