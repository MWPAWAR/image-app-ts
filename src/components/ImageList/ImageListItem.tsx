import { useAppDispatch } from '../customHooks/hooks'
import { getMbsTextByBytes } from '../../utils/get-mbs-by-bytes'
import imageListItemStyles from './ImageListItem.module.css'
import { setSelectedImg } from '../../store/imageUiState/imageUiState'
import { ImageListItemProps } from './types'

const ImageListItem = ({ image }: ImageListItemProps) => {
  const dispatch = useAppDispatch()
  const handleImageClick = (): void => {
    dispatch(setSelectedImg({ id: image.id }))
  }

  return (
    <div
      data-testid="image-list-item"
      onClick={handleImageClick}
      className={imageListItemStyles.imageListItem}
    >
      <img
        tabIndex={0}
        src={image.url}
        className={imageListItemStyles.imageListItemImg}
        alt={image.filename}
      />
      <div className={imageListItemStyles.imageListItemName}>
        {image.filename}
      </div>
      <div className={imageListItemStyles.imageListItemSize}>
        {getMbsTextByBytes(image.sizeInBytes)}
      </div>
    </div>
  )
}

export default ImageListItem
