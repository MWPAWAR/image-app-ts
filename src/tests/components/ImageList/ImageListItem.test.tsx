import { MemoryRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import renderer, { ReactTestRenderer } from 'react-test-renderer'
import configureStore, { MockStore, MockStoreCreator } from 'redux-mock-store'
import { render, RenderResult } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import ImageListItem from '../../../components/ImageList/ImageListItem'
import { imageMock1 } from '../../mocks/image'
import { toggleDrawer } from '../../test-utils'

describe('ImageListItem', () => {
  const mockStore: MockStoreCreator = configureStore([])
  const store: MockStore = mockStore({})
  jest.spyOn(store, 'dispatch')

  const ImageListItemComponent = () => (
    <MemoryRouter>
      <Provider store={store}>
        <ImageListItem image={imageMock1} toggleDrawer={toggleDrawer} />
      </Provider>
    </MemoryRouter>
  )

  it('Matches snapshot', () => {
    const tree: ReactTestRenderer = renderer.create(<ImageListItemComponent />)
    expect(tree).toMatchSnapshot()
  })

  it('calls dispatch if user clicks on ImageListItem', async () => {
    const { getByTestId }: RenderResult = render(<ImageListItemComponent />)
    await userEvent.click(getByTestId('image-list-item'))
    expect(store.dispatch).toBeCalled()
    expect(store.dispatch).toHaveBeenCalledWith({
      type: 'imageUiState/setSelectedImg',
      payload: { id: imageMock1.id },
    })
  })
})
