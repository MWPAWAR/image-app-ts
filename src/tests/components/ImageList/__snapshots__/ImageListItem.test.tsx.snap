// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageListItem Matches snapshot 1`] = `
<div
  className="imageListItem"
  data-testid="image-list-item"
  onClick={[Function]}
>
  <img
    alt="tennessee_female_rubber.jpg"
    className="imageListItemImg"
    src="https://agencyanalytics-api.vercel.app/images/0.jpg"
    tabIndex={0}
  />
  <div
    className="imageListItemName"
  >
    tennessee_female_rubber.jpg
  </div>
  <div
    className="imageListItemSize"
  >
    4.6 MB
  </div>
</div>
`;
